File: Petstore-Load-Test/userTests.py
from locust import HttpUser, between, task

class WebsiteUser(HttpUser):
    wait_time = between(5, 15)

    def createUserFunction(self, id, username):
        payload = {
            "id": id,
            "username": username,
            "firstName": "test87",
            "lastName": "example",
            "email": "api80@gmail.com",
            "password": "111",
            "phone": "5064732018",
            "userStatus": 1
        }
        self.client.post("/v2/user", json=payload)

    @task
    def deleteUser(self):
        self.createUserFunction(9090, "api80")
        self.client.delete("/v2/user/api80")

    @task
    def createUser(self):
        self.createUserFunction(85,"successful")

    @task
    def getUserInfo(self):
        self.createUserFunction(9090, "api80")
        self.client.get("/v2/user/api80")

    @task
    def login(self):
        self.client.get("/v2/user/login?username=load&password=98765")

    @task
    def logout(self):
        self.client.get("/v2/user/logout")    

    @task
    def uptadeUser(self):
        body = {
            "id": 8080,
            "username": "api",
            "firstName": "test",
            "lastName": "example",
            "email": "api@gmail.com",
            "password": "7438",
            "phone": "5064793618",
            "userStatus": 0
        }
        self.client.put("/v2/user/load", json=body)
    

File: Peststore-Api-Automation/tests/userTests.js
const chakram = require('chakram'), expect = chakram.expect;

describe("User Module", function() {

    it("GET - GetUserInfo", function () {
        const resBody = {
            "id": 7448,
            "username": "test89",
            "firstName": "ali",
            "lastName": "ulker",
            "email": "ali@gmail.com",
            "password": "09876",
            "phone": "4628692748",
            "userStatus": 0
         }

         const response = chakram.get("https://petstore.swagger.io/v2/user/test89");
         expect(response).to.have.status(200);
         expect(response).to.comprise.of.json(resBody);
         expect(response).to.have.header('content-type', 'application/json'); 
         return chakram.wait();
      });  
   
    it("DELETE - DeleteUser", function () {
        const resBody = {
            "code": 200,
            "type": "unknown",
            "message": "test89"
         };

         const response = chakram.delete("https://petstore.swagger.io/v2/user/test89");
         expect(response).to.have.status(200);
         expect(response).to.comprise.of.json(resBody); 
         expect(response).to.have.header('content-type', 'application/json'); 
         return chakram.wait();
      });

   
    it("GET - GetUserInfo Invalid User", function () {
        const resBody = {
            "code": 1,
            "type": "error",
            "message": "User not found"
         };

         const response = chakram.get("https://petstore.swagger.io/v2/user/test89");
         expect(response).to.have.status(404);
         expect(response).to.comprise.of.json(resBody); 
         expect(response).to.have.header('content-type', 'application/json'); 
         return chakram.wait();
      });

    it("POST - CreateUser", function () {
        const resBody = {
            "code": 200,
            "type": "unknown",
            "message": "7448"
         };

         const body = {
             "id": 7448,
             "username": "test89",
             "firstName": "ali",
             "lastName": "ulker",
             "email": "ali@gmail.com",
             "password": "09876",
             "phone": "4628692748",
             "userStatus": 0
         };

         const response = chakram.post("https://petstore.swagger.io/v2/user", body);
         expect(response).to.have.status(200);
         expect(response).to.comprise.of.json(resBody); 
         expect(response).to.have.header('content-type', 'application/json'); 
         return chakram.wait();
      });

   

    it("PUT - UpdateUser", function () {
        const resBody = {
            "code": 200,
            "type": "unknown",
            "message": "1004"
         };

         const body = {
             "id": 1004,
             "username": "test12345",
             "firstName": "ahmet",
             "lastName": "yılmaz",
             "email": "test@gmail.com",
             "password": "7890",
             "phone": "4789364736",
             "userStatus": 0
         }

         const response = chakram.put("https://petstore.swagger.io/v2/user/test123", body);
         expect(response).to.have.status(200);
         expect(response).to.comprise.of.json(resBody); 
         expect(response).to.have.header('content-type', 'application/json'); 
         return chakram.wait();
      });
});